pennylane/data/attributes/list.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains an AttributeType that allows for heterogenous lists of dataset
+types."""
+
+import typing
+from collections.abc import MutableSequence, Sequence
+from typing import Generic, List, Union, overload
+
+from pennylane.data.base.attribute import AttributeType
+from pennylane.data.base.mapper import MapperMixin
+from pennylane.data.base.typing_util import HDF5Any, HDF5Group, T
+
+
+class DatasetList(
+    Generic[T],
+    AttributeType[HDF5Group, typing.Sequence[T], typing.Iterable[T]],
+    MutableSequence,
+    MapperMixin,
+):
+    """Provides a list-like collection type for dataset attributes."""
+
+    type_id = "list"
+
+    def __post_init__(self, value: typing.Iterable[T], info):
+        self.extend(value)
+
+    def default_value(self) -> typing.Iterable[T]:
+        return ()
```

Comment:
Does it make more sense for this to be `[]` since it is mutable?

Reply:
sure